src/app-interoperability/application-package-definition.linkml.yaml

```
@@ -0,0 +1,497 @@
+      settings:
+        description: >-
+          Settings are used to provide instructions to the workload orchestration software vendor for displaying parameters to the user. 
+          A user MUST be able to provide values for all settings. 
+          See the [Setting](#setting-attributes) section below.
+        rank: 20
+        range: Setting
+        multivalued: true
+        required: true
+
+  Setting:
+    description: Individual configuration settings.
+    rank: 130
+    attributes:
+      parameter:
+        description: The name of the [parameter](#parameter-attributes) the setting is associated with.
+        rank: 10
+        range: string
+        required: true
+      name:
+        description: The parameter's display name to show in the user interface.
+        rank: 20
+        range: string
+        required: true
+      description:
+        description: The parameters's short description to provide additional context to the user in the user interface about what the parameter is for.
+        rank: 30
+        range: string
+      immutable:
+        description: If true, indicates the parameter value MUST not be changed once it has been set and used to install the application. Default is false if not provided.
+        rank: 40
+        range: boolean
+      schema:
+        description: The name of the schema definition to use to validate the parameter's value. See the [Schema](#schema-attributes) section below.
+        rank: 50
+        range: string
+        required: true
+
+  Schema:
+    description: Validation rules for parameter values.
+    rank: 140
+    attributes:
+      name:
+        description: The name of the schema rule. This used in the [setting](#setting-attribute) to link the setting to the schema rule.
+        rank: 10
+        range: string
+        required: true
+      datatype:
+        description: >-
+          Indicates the expected data type for the user provided value. 
+          Accepted values are string, integer, double, boolean, array[string], array[integer], array[double], array[boolean]. 
+          At a minimum, the provided parameter value MUST match the schema's data type if no other validation rules are provided.
+        rank: 20
+        range: string
+        required: true
+        pattern: ^(string|integer|double|boolean|array\[.*\])$
+      #Arne: error in specification: 'validation rule' not accurate. 
+      
+      # Text Validation Rules
+      allowEmpty:
+        description: If true, indicates a value must be provided. Default is false if not provided.
+        rank: 30
+        range: boolean
+      minLength:
+        description: If set, indicates the minimum number of characters the value must have to be considered valid.
+        rank: 40
+        range: integer
+      maxLength:
+        description: If set, indicates the maximum number of characters the value must have to be considered valid.
+        rank: 50
+        range: integer
```

@arne-broering missing `schema` attributes

```suggestion
        range: integer
      minValue:
        description: If set, indicates the minimum value to be considered valid.
        rank: 50
        range: float
      maxValue:
        description: If set, indicates the maximum value to be considered valid.
        rank: 50
        range: float
      maxPrecision:
        description: If set, indicates the maximum precision to be considered valid.
        rank: 50
        range: integer
```